### PR TITLE
Imx8mm ca53

### PR DESCRIPTION
**Pull request by MrVan**
This is to support i.MX8MM CA53 build for Zephyr. Otherwise, upstream zephyr CI will trigger build error for https://github.com/zephyrproject-rtos/zephyr/pull/32694

Most code are copied/modied from M4 side and adapt for A53 usage.

### CODE REVIEW

**Comment by MrVan** [comment]
@MaureenHelm

**Comment by MrVan** [comment]
@MaureenHelm @galak I updated this PR based on the #107. The mcux-sdk has been pushed to feature/aarch64_support, for long term, that will be moved to master.

**Comment by MrVan** [comment]
@MaureenHelm @galak Rebased to master branch. Please check.

File: mcux/devices/MIMX8MM6/core_ca53.h [hunk 1]
```diff
@@ -0,0 +1,855 @@
...
+   LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+   CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+   SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+   INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+   CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+   ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+   POSSIBILITY OF SUCH DAMAGE.
+   ---------------------------------------------------------------------------*/
+
+#if   defined ( __ICCARM__ )
```
Comment (MaureenHelm) [inline mcux/devices/MIMX8MM6/core_ca53.h:29]: Some of these files look like they come from CMSIS. Why do we need copies of them in the hal_nxp module?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please update `mcux/README` to add this device and document its MCUXpresso SDK version.

File: mcux/devices/MIMX8MM6/core_ca53.h (hunk 1 above)
Comment (MrVan) [reply mcux/devices/MIMX8MM6/core_ca53.h:29]: CMSIS not support ARMv8-A. I'll update the header file.

**Comment by MrVan** [comment]
> Please update `mcux/README` to add this device and document its MCUXpresso SDK version.
> 
> Also, it doesn't look like your PR in the main zephyr references this one. Would like to see CI pass in the main zephyr repo before merging this hal_nxp PR.

Could you please share how to refer this PR in zephyr PR and make CI pass? In my local, I have changed my code to use hal/nxp.

**Comment by MaureenHelm** [comment]
> > Also, it doesn't look like your PR in the main zephyr references this one. Would like to see CI pass in the main zephyr repo before merging this hal_nxp PR.
> 
> Could you please share how to refer this PR in zephyr PR and make CI pass? In my local, I have changed my code to use hal/nxp.

https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-module

File: mcux/devices/MIMX8MM6/core_ca53.h (hunk 1 above)
Comment (mmahadevan108) [reply mcux/devices/MIMX8MM6/core_ca53.h:29]: Can you take a look at this and see if this has what you need. https://github.com/zephyrproject-rtos/cmsis/tree/master/CMSIS/Core_A/Include

**Comment by MrVan** [comment]
Ping.. It's been long time the PR pending here. I wonder if this is not acceptable for you(owner of hal_nxp), I will totally remove the dependency again and only support jailhouse virtualization enviorment in my zephyr OS PR.

**Comment by dleach02** [comment]
@MrVan is this PR still relevant?

**Comment by dleach02** [comment]
Closing. Can be reopened if need arises.

**Pull request closed without merging**